### PR TITLE
chore(symdb): handle potential import race on enable

### PR DESCRIPTION
**Pull request by P403n1x87**
Because the Symbol DB uploads are enabled via RC signals, any imports happening in parallel with Symbol DB enablement might get missed. We ensure that all modules are processed by iterating over all modules once more after enablement.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/internal/symbol_db/symbols.py [hunk 1]
```diff
@@ -531,10 +531,32 @@ class SymbolDatabaseUploader(BaseModuleWatchdog):
...
         # Look for all the modules that are already imported when this is
         # installed and upload the symbols that are marked for inclusion.
         context = ScopeContext()
-        for module in (_ for _ in list(sys.modules.values()) if is_module_included(_)):
+        for name, module in list(sys.modules.items()):
+            # Skip modules that are being initialized as they might not be
+            # fully loaded yet.
+            try:
+                if module.__spec__._initializing:  # type: ignore[union-attr]
+                    continue
```
Comment (ojung) [inline ddtrace/internal/symbol_db/symbols.py:548]: Should we retry them?

Comment (P403n1x87) [reply ddtrace/internal/symbol_db/symbols.py:548]: That's what the extra update will do 🙂

**Review by ojung — APPROVED** [review]

File: ddtrace/internal/symbol_db/symbols.py (hunk 1 above)
Comment (shatzi) [reply ddtrace/internal/symbol_db/symbols.py:548]: could modules get loaded after RC config upgrade was received? an application load more modules dynamically over time?

**Review by shatzi — APPROVED** [review]

File: ddtrace/internal/symbol_db/symbols.py (hunk 1 above)
Comment (P403n1x87) [reply ddtrace/internal/symbol_db/symbols.py:548]: For those we already have the import hook. Here we want to make sure that we catch those that might have been imported while we are installing the uploader

**Pull request merged**